### PR TITLE
Ticket 7 updates

### PR DESCRIPTION
**Pull request by han-lawrence**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -220,12 +220,62 @@ describe('7. POST /api/articles/:article_id/comments', () => {
 	.then(({ body }) => {
 	expect(body).toEqual({ msg: 'Incorrect File Path' });
 		});
+});
 
 
+test(' returns a 400 if the data to post is missing body.', () => {
+	const item = { username: 'Hannah77' };
```
Comment (callumu21) [inline __tests__/app.test.js]: Be careful with these tests.  A post is only valid if the username is included in the database, which 'Hannah77' isn't. As you're testing various other conditions, we should be using a valid username in these tests

File: __tests__/app.test.js [hunk 2]
```diff
@@ -272,4 +322,36 @@ test ('returns a 400 if inc_votes is missing.', () => {
         });
 });
 
+
+test('returns a 404 found if the username is not found.', () => {
```
Comment (callumu21) [inline __tests__/app.test.js:359]: Should these tests be in this describe block? It looks like you have a test for PATCH requests above this, so you may need to rejig some of these around.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -272,4 +322,36 @@ test ('returns a 400 if inc_votes is missing.', () => {
...
+				};
+        return request(app)
+        .post('/api/articles/3/comments')
+        .send(item)
+        .expect(404)
+        .then(({body}) => expect(body.msg).toBe('Username not found'));
+
+});
+
+test('returns a 404 if no article matches ID.', () => {
```
Comment (callumu21) [inline __tests__/app.test.js]: This test looks identical to the one on line 247, so not too sure that it's still needed.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -220,12 +220,62 @@ describe('7. POST /api/articles/:article_id/comments', () => {
...
+			expect(msg).toBe('ID not found');
+		});
+});
+
+
+
+});
+
+
+
```
Comment (callumu21) [inline __tests__/app.test.js:300]: Just missing two more tests:

1. The POST body is correct, but also has unnecessary properties on it. This should still be successful, but you'll want to make sure any extra properties are just ignored.
2. What happens if the article_id is not valid (e.g. /api/articles/banana/comments)?

**Review by callumu21 — CHANGES_REQUESTED** [review]
Nice start here; just missing a couple more tests as per my comment below.

It also looks like you've added some tests into the wrong describe block, so you will want to move these around. Also, make sure that the post body for all tests is the same (e.g. {username: 'butter_bridge', body: 'This is a body'}) unless you are deliberately testing that the body is incorrect. Otherwise, you won't know if it's failing because the body is wrong, or if it's because of the condition you're actually testing for.

Once you've made those changes, please resubmit this PR for review 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Hannah. Thanks for adding the invalid ID test. We seem to just be missing the first one that Callum asked for though.
We just need one more 201 test which has an extra property in the post object. So it would have username, body and then something else random (banana etc) and this would be ignored. Could you just add this test please? 🙂

**Pull request merged**